### ISSUE TITLE
nullable field isn't nullable if also restricted to a list of choices

### ISSUE THREAD

**Issue by classabbyamp** [collerek/ormar#354]
## Describe the bug
Fields with `nullable=True` aren't considered nullable when also using `choices=...`

## To Reproduce
Create a model with the following:
```py
from enum import Enum
from ormar import Model, Integer, String
import sqlalchemy
import databases

URL = "postgresql://..."  # url to postgres db

metadata = sqlalchemy.MetaData()
database = database.Database(URL)

class MyChoices(Enum):
    A = "A"
    B = "B"

class Item(Model):
    class Meta:
        tablename="items"
        metadata = metadata
        database = database

    id = Integer(primary_key=True)
    my_field = String(max_length=1, nullable=True, choices=list(MyChoices))

app = FastAPI()
app.state.database = database

@app.on_event("startup")
async def startup() -> None:
    database_ = app.state.database
    if not database_.is_connected:
        await database_.connect()


@app.on_event("shutdown")
async def shutdown() -> None:
    database_ = app.state.database
    if database_.is_connected:
        await database_.disconnect()

@app.post("/upload/")
async def upload(item: Item):
    await Item.save()
```

Upon trying a POST request to that endpoint, if `my_field` is set to `None`, it will fail with a ValueError:

```json
{"detail": [{"loc": ["body", 0, "__root__"], "msg": "my_field: 'None' not in allowed choices set: ['A', 'B']", "type": "value_error"}]}
```

## Expected behavior

The model accepts `"A"`, `"B"`, or `None` for `my_field`, because it has a set of valid choices and is nullable. It can be worked around by adding `None` to the enum, but that does not seem natural to me:

```diff
  class MyChoices(Enum):
      A = "A"
      B = "B"
+     NONE = None
```

## Versions
 - Database backend used: postgres
 - Python version: 3.9.7
 - `ormar` version: 0.10.19
 - `pydantic` version: 1.8.2
 - `fastapi` version: 0.68.1

**Comment by collerek** [comment]
That's the same issue as described in #353, i will introduce the extra parameter on Model meta, that will allow to ignore or forbid the extra fields

**Comment by classabbyamp** [comment]
I don't think it is the same issue. I am only sending it the required/allowed fields, but the `choices=` restriction on the nullable field isn't accepting `None` unless I put `None` in my list of choices

**Comment by collerek** [comment]
Yep you are right, nullable should still cause the field to be optional. Thank you for catching that!

**Comment by collerek** [comment]
> I don't think it is the same issue. I am only sending it the required/allowed fields, but the `choices=` restriction on the nullable field isn't accepting `None` unless I put `None` in my list of choices

Sorry last post was in wrong issue 🤦

**Comment by classabbyamp** [comment]
oh no problem, thanks!

**Comment by collerek** [comment]
Fixed in 0.10.20
